src/UI/Components/InstanceProvider/InstanceProvider.tsx

```diff
@@ -1,34 +1,79 @@
 import React, { useContext } from "react";
 import { ServiceModel } from "@/Core";
-import { DependencyContext } from "@/UI";
-import { RemoteDataView } from "@/UI/Components";
+import { useGetInstanceWithRelations } from "@/Data/Managers/V2/GetInstanceWithRelations";
+import { useGetMetadata } from "@/Data/Managers/V2/GetMetadata";
+import { DependencyContext, words } from "@/UI";
+import { ErrorView, LoadingView } from "@/UI/Components";
 import Canvas from "@/UI/Components/Diagram/Canvas";
 
+/**
+ * Renders the InstanceProvider component.
+ * It serves purpose to not provide all the neccesary data for the Canvas at the same time to avoid unnecessary rerenders,
+ *  and extract the data fetching logic, out of the already busy component
+ *
+ * @param {ServiceModel[]} services - The list of service models.
+ * @param {string} mainServiceName - The name of the main service.
+ * @param {string} instanceId - The ID of the instance.
+ * @returns {JSX.Element} The rendered InstanceProvider component.
+ */
 export const InstanceProvider: React.FC<{
   label: string;
   services: ServiceModel[];
   mainServiceName: string;
   instanceId: string;
   editable?: boolean;
-}> = ({ label, services, mainServiceName, instanceId, editable = false }) => {
-  const { queryResolver } = useContext(DependencyContext);
-  const [data] = queryResolver.useOneTime<"GetInstanceWithRelations">({
-    kind: "GetInstanceWithRelations",
-    id: instanceId,
-  });
+}> = ({ services, mainServiceName, instanceId, editable = false }) => {
+  const { environmentHandler } = useContext(DependencyContext);
+  const environment = environmentHandler.useId();
+
+  const { data, isError, isLoading, error, refetch } =
```

Comment:
Perhaps it would have been more readable to not opt for deconstruction when having two React Queries in the same component.

It would avoid confusion, because now, you're explicitly renaming the metadata query deconstruction, but not the one for the instance.

I had the same issue in my current wip and solved it like this to keep it neat.
<img width="278" alt="image" src="https://github.com/inmanta/web-console/assets/44098050/aa714a6b-18c7-4483-9656-79d4b65593d2">